consensus/vbft/service.go

```diff
@@ -626,14 +625,15 @@ func (self *Server) getState() ServerState {
 
 func (self *Server) updateParticipantConfig() error {
 	blkNum := self.GetCurrentBlockNo()
-
 	block, _ := self.blockPool.getSealedBlock(blkNum - 1)
 	if block == nil {
 		return fmt.Errorf("failed to get sealed block (%d)", blkNum-1)
 	}
-
 	var err error
 	self.metaLock.Lock()
+	if self.checkNeedUpdateChainConfig(blkNum) || self.checkUpdateChainConfig() {
```

Comment:
do this check before metaLock.Lock() ?

Reply:
btw, 
always self.config = block.Info.NewChainConfig when block.Info.NewChainConfig is not nil ?